modules/gapi/src/backends/ie/giebackend.cpp

```diff
@@ -217,6 +217,11 @@ struct IEUnit {
         if (params.num_out == 1u && params.output_names.empty()) {
             params.output_names = { outputs.begin()->first };
         }
+        // Reshape for configured map with layers' names and their new shapes
+        if (!params.reshape_inputs.empty()) {
+            GAPI_Assert(params.reshape_inputs.size() <= params.num_in);
+            net.reshape(params.reshape_inputs);
```

Comment:
So how it works for your purpose? Why we're not looking at the input _data_ dimensions? Isn't this the case?